app/__tests__/src/pages/Login.test.js

```diff
@@ -0,0 +1,94 @@
+import React from "react";
+import renderer from 'react-test-renderer';
+import { MemoryRouter } from "react-router-dom"; 
+import { describe, it, expect } from '@jest/globals';
+import Login from '../../../src/pages/Login';
+import axios from 'axios';
+import MockAdapter from'axios-mock-adapter';
+import AuthServices from '../../../src/services/auth.services'
+
+jest.useFakeTimers();
+
+describe("Login Component", () => {
+
+it("can render login page without errors", async () => {
+    let component;
+    await renderer.act(async () => {
+        component = renderer.create(
+        <MemoryRouter>
+            <Login />
+        </MemoryRouter>
+    )});
+
+    expect(component.toJSON()).toMatchSnapshot();
+});
+
+it("can navigate to the welcome page and the signup pages", async () => {
+    let component;
+    await renderer.act(async () => {
+        component = renderer.create(
+        <MemoryRouter>
+            <Login />
+        </MemoryRouter>
+    )});
+
+    const entrarButton = component.root.findAllByProps({ to: "/welcome" })[1];
+    const submitButton = component.root.findByProps({ to: "/signup" });
+
+    expect(entrarButton && submitButton).toBeTruthy();
+    });
+
+  it("can submit a correct HTTP request", async () => {
```

Comment:
This test doesn't seem to test your page - it tests the postUserLogin function in auth.services

Reply:
This reply is relevant for the comment below as well.
We currently have not found a way to test a function that is internal to a component, so the second best option for Login and Signup was to make sure that they could send HTTP messages, which is in principle and external function not necessarily relevant for the testing of pages themselves.